src/openrct2-ui/windows/SavePrompt.cpp

```diff
@@ -75,13 +75,46 @@ static void WindowSavePromptCallback(int32_t result, const utf8* path)
     }
 }
 
-class SavePromptWindow final : public rct_window
+class SavePromptWindow final : public Window
 {
 public:
-    /**
-     *
-     *  rct2: 0x0066DF17
-     */
+    void OnOpen() override
+    {
+        PromptMode prompt_mode;
+        StringId stringId;
+
+        if (gScreenFlags & (SCREEN_FLAGS_TRACK_DESIGNER | SCREEN_FLAGS_TRACK_MANAGER))
+        {
+            widgets = window_quit_prompt_widgets;
+        }
+        else
+        {
+            widgets = window_save_prompt_widgets;
+        }
+        InitScrollWidgets();
+
+        // Pause the game if not network play.
+        if (network_get_mode() == NETWORK_MODE_NONE)
+        {
+            gGamePaused |= GAME_PAUSED_MODAL;
+            OpenRCT2::Audio::StopAll();
+        }
+
+        window_invalidate_by_class(WindowClass::TopToolbar);
+
+        prompt_mode = gSavePromptMode;
+        if (prompt_mode == PromptMode::Quit)
+            prompt_mode = PromptMode::SaveBeforeQuit;
```

Comment:
You already have the `prompt_mode` set when you create the window, and it should be set on the class constructor to avoid reading from a global. I'll show below how it can be done.